packages/wrangler/src/versions/secrets/bulk.ts

```diff
@@ -0,0 +1,128 @@
+import path from "node:path";
+import readline from "node:readline";
+import { fetchResult } from "../../cfetch";
+import { readConfig } from "../../config";
+import { UserError } from "../../errors";
+import { getLegacyScriptName } from "../../index";
+import { logger } from "../../logger";
+import { parseJSON, readFileSync } from "../../parse";
+import { validateJSONFileSecrets } from "../../secret";
+import { printWranglerBanner } from "../../update-check";
+import { requireAuth } from "../../user";
+import { copyWorkerVersionWithNewSecrets } from "./index";
+import type {
+	CommonYargsArgv,
+	StrictYargsOptionsToInterface,
+} from "../../yargs-types";
+import type { WorkerVersion } from "./index";
+
+export function versionsSecretsPutBulkOptions(yargs: CommonYargsArgv) {
+	return yargs
+		.positional("json", {
+			describe: `The JSON file of key-value pairs to upload, in form {"key": value, ...}`,
+			type: "string",
+		})
+		.option("name", {
+			describe: "Name of the Worker",
+			type: "string",
+			requiresArg: true,
+		})
+		.option("message", {
+			describe: "Description of this deployment (optional)",
+			type: "string",
+			requiresArg: true,
+		})
+		.option("tag", {
+			describe: "A tag for this version (optional)",
+			type: "string",
+			requiresArg: true,
+		});
+}
+
+export async function versionsSecretPutBulkHandler(
+	args: StrictYargsOptionsToInterface<typeof versionsSecretsPutBulkOptions>
+) {
+	await printWranglerBanner();
+	const config = readConfig(args.config, args, false, true);
+
+	const scriptName = getLegacyScriptName(args, config);
+	if (!scriptName) {
+		const error = new UserError(
+			"Required Worker name missing. Please specify the Worker name in wrangler.toml, or pass it as an argument with `--name <worker-name>`"
+		);
+		logger.error(error.message);
```

Comment:
I'm not sure we need to both log **and** throw the error. I think throwing should be enough?

Reply:
ha yep indeed, this is weird. This is a copy-pasta from the regular secret bulk

Reply:
I **think**
```
throw new FatalError("Required Worker name missing. Please specify the Worker name in wrangler.toml, or pass it as an argument with `--name <worker-name>`")
```
should do it

Reply:
FatalError would try and log to Sentry, UserError should be enough no?

Reply:
oh yeah, you're right 👍